app/models/nic/interface.rb

```diff
@@ -23,7 +23,10 @@ class Interface < Base
     delegate :network, :to => :subnet
 
     def vlanid
-      self.tag.blank? ? self.subnet.vlanid : self.tag
+      return self.tag unless self.tag.blank?
+      return self.subnet.vlanid if self.subnet
+      return self.subnet6.vlanid if self.subnet6
```

Comment:
if `self.subnet6` is present but `self.subnet6.vlanid` is blank, it will return '' instead of nil, it should also probably be a part of a condition